src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1211,30 +1211,34 @@ void DoCancelRequest(int backendProcessId, int backendSecretKey)
 
         #region Close / Reset
 
+        internal bool HasOngoingOperation => CurrentReader != null || CurrentCopyOperation != null;
+
         /// <summary>
         /// Closes ongoing operations, i.e. an open reader exists or a COPY operation still in progress, as
         /// part of a connection close.
         /// Does nothing if the thread has been aborted - the connector will be closed immediately.
         /// </summary>
-        internal void CloseOngoingOperations()
+        internal async Task CloseOngoingOperations(bool async)
         {
-            CurrentReader?.Close(true, false);
-            var currentCopyOperation = CurrentCopyOperation;
-            if (currentCopyOperation == null)
+            var (reader, copyOperation) = (CurrentReader, CurrentCopyOperation);
```

Comment:
You really find this code confusing? Maybe I'm used to it from Python or something, but it seems very clear and terse to me. If you insist I'll remove it.

I touched this function in general in order to make it async, because of the closing of the reader, which can mean consuming a large pending resultset.

Reply:
> You really find this code confusing?

In .NET world it isn't to clear. Personally I would like to follow common C# code style like other projects in the community (with some exclusions maybe), because it's more usual for people in this area. Therefore, it's easier to contribute.

> I touched this function in general in order to make it async

I mean what is a reason to save field values to variables. Is it because of concurrency? The reader isn't used in the code below, and as you know the call can be simplified.

Reply:
The capturing of fields to local variables made sense in a previous version of this PR, reverted.

Reply:
Reverted the Python-style assignments.

Re the capturing field values to variables, this is important because of async/await. We can't write `await reader?.Close()`, because if reader is null we get an NRE. This may also become more important in the future if we start handling various race conditions when closing an in-use connection.